smartsim/experiment.py

```diff
@@ -797,6 +796,56 @@ def reconnect_orchestrator(self, checkpoint: str) -> Orchestrator:
             logger.error(e)
             raise
 
+    def preview(
+        self,
+        output_format: t.Optional[str] = None,
+        output_filename: t.Optional[str] = None,
+        verbosity_level: t.Optional[str] = None,
+    ) -> str:
+        """Preview entity information prior to launch. This method
+        aggregates multiple pieces of information to give users insight
+        into what and how entities will be launched.  Any instance of
+        ``Model``, ``Ensemble`` or ``Orchestrator`` created by the
+        Experiment can be passed as an argument to the preview method.
+        param output_format: Set output destination. The possible accepted
+        output formats are `json`, `xml`, `html`, `plain_text`, `color_text`.
+        A filename is required if an output format is specified. If no output
+        format is set, the preview will be output to stdout. Defaults to None.
+        type output_type: str
+        param output_filename: Specify name of path to write preview data to.
+        Only needed when an output format has been specified. Defaults to None.
+        type output_filename: str
+        param verbosity level: the verbosity level.
+        info: Display User defined fields and entities
+        debug: Display user defined field and entities and auto generated
+        fields.
+        developer: Display user defined field and entities, auto generated
+        fields, and run commands.
+        Defaults to info.
+        type verbosity_level: str
+        """
+
+        if output_format:
+            raise NotImplementedError
+        if output_filename:
+            raise NotImplementedError
+        if verbosity_level:
+            raise NotImplementedError
+
+        return self._preview()
+
+    def _preview(self) -> str:
```

Comment:
Consider renaming `ViewExp` to `ExpPreviewRenderer` to clearly indicate its purpose

Consider getting rid of the `ViewExp` class altogether and just using a function since no members are really used. If not, consider static methods so you can get rid of `Experiment._preview` and use:

```py
return ExpView.render(self)
```